src/main/java/org/myrobotlab/service/LcdDisplay.java

```diff
@@ -0,0 +1,232 @@
+  private final byte LCD_ENTRYSHIFTDECREMENT = (byte) 0x00;
+
+  // flags for display on/off control
+  private final byte LCD_DISPLAYON = (byte) 0x04;
+  private final byte LCD_DISPLAYOFF = (byte) 0x00;
+  private final byte LCD_CURSORON = (byte) 0x02;
+  private final byte LCD_CURSOROFF = (byte) 0x00;
+  private final byte LCD_BLINKON = (byte) 0x01;
+  private final byte LCD_BLINKOFF = (byte) 0x00;
+
+  // flags for display/cursor shift
+  private final byte LCD_DISPLAYMOVE = (byte) 0x08;
+  private final byte LCD_CURSORMOVE = (byte) 0x00;
+  private final byte LCD_MOVERIGHT = (byte) 0x04;
+  private final byte LCD_MOVELEFT = (byte) 0x00;
+
+  // flags for function set
+  private final byte LCD_8BITMODE = (byte) 0x10;
+  private final byte LCD_4BITMODE = (byte) 0x00;
+  private final byte LCD_2LINE = (byte) 0x08;
+  private final byte LCD_1LINE = (byte) 0x00;
+  private final byte LCD_5x10DOTS = (byte) 0x04;
+  private final byte LCD_5x8DOTS = (byte) 0x00;
+
+  // flags for backlight control
+  private final byte LCD_BACKLIGHT = (byte) 0x08;
+  private final byte LCD_NOBACKLIGHT = (byte) 0x00;
+
+  private final byte En = (byte) 0b00000100; // Enable bit
+  private final byte Rw = (byte) 0b00000010; // Read/Write bit
+  private final byte Rs = (byte) 0b00000001; // Register select bit
+
+  private boolean backLight;
+  public Map<Integer, String> screenContent = new HashMap<Integer, String>();
+
+  public void attach(Pcf8574 pcf8574) {
+    // we need more checkup here...
+    if (pcf8574.isAttached) {
+      this.i2cControler = pcf8574;
+      setReady(true);
+      log.info("{} attach {}", getName(), pcf8574.getName());
+    } else {
+      log.error("Cannot attach {} to {}", getName(), pcf8574.getName());
+    }
+  }
+
+  /*
+   * INIT LCD panel
+   */
+  public void init() {
+    log.info("Init I2C Display");
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x02);
+    lcd_write((byte) (LCD_FUNCTIONSET | LCD_2LINE | LCD_5x8DOTS | LCD_4BITMODE));
+    lcd_write((byte) (LCD_DISPLAYCONTROL | LCD_DISPLAYON));
+    lcd_write((byte) (LCD_CLEARDISPLAY));
+    lcd_write((byte) (LCD_ENTRYMODESET | LCD_ENTRYLEFT));
+  }
+
+  /*
+   * Send byte to PCF controller
+   */
+  public void write_cmd(byte cmd) {
+    if (isReady()) {
+      i2cControler.writeRegister(cmd);
+    } else {
+      log.error("LCD is not ready / attached !");
+    }
+  }
+
+  /*
+   * Turn ON/OFF LCD backlight
+   */
+  public void setBackLight(boolean status) {
+    if (status == true) {
+      write_cmd(LCD_BACKLIGHT);
+    } else {
+      write_cmd(LCD_NOBACKLIGHT);
+    }
+    backLight = status;
+    log.info("LCD backlight set to {}", status);
+    broadcastState();
+  }
+
+  public boolean getBackLight() {
+    return backLight;
+  }
+
+  /*
+   * Display string on LCD, by line 
+   */
+  public void display_string(String string, int line) {
+    screenContent.put(line, string);

```

Comment:
heh ... this is Python style .. remember were in Java-Land now ;) should be displayString(String string, int line) or perhaps since the type is string (and its obvious a string) then just display(String string, int line) ...

Reply:
ohh yeh 👍  tks